tests/e2e/scale/test_scale_osds_fill_75%_reboot_workers.py

```diff
@@ -0,0 +1,108 @@
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources.pod import wait_for_dc_app_pods_to_reach_running_state
+from ocs_ci.ocs.resources import storage_cluster
+from ocs_ci.framework.testlib import scale, E2ETest, ignore_leftovers, tier4a
+from ocs_ci.utility.utils import ceph_health_check
+from tests.conftest import nodes
+
+
+logger = logging.getLogger(__name__)
+
+
+@scale
+@tier4a
+@ignore_leftovers
+@pytest.mark.parametrize(
+    argnames=["interface"],
+    argvalues=[
+        pytest.param(constants.CEPHBLOCKPOOL),
+        pytest.param(constants.CEPHFILESYSTEM)
+    ]
+)
+class TestScaleOSDsRebootNodes(E2ETest):
+    """
+    Add first set of OSD to a minimum cluster with 50%
+    of storage utilization and wait for rebalance
+    Reboot worker nodes after rebalance
+    """
+    num_of_pvcs = 10
+    pvc_size = 5
+
+    def test_scale_osds_reboot_nodes(self, interface, project_factory, multi_pvc_factory, dc_pod_factory):
+        """
+        Check storage utilization, if its less then runs IO,
+        Scale osds from 3-6, check for rebalance and reboot workers
+        """
+        current_osd_count = count_cluster_osd()
+        proj_obj = project_factory()
+        if current_osd_count == 3:
+            while not validate_osd_utilization(osd_used=50):
+
+                # Create pvc
+                pvc_objs = multi_pvc_factory(
+                    project=proj_obj,
+                    interface=interface, size=self.pvc_size,
+                    num_of_pvc=self.num_of_pvcs
+                )
+
+                dc_pod_objs = list()
+                for pvc_obj in pvc_objs:
+                    dc_pod_objs.append(dc_pod_factory(pvc=pvc_obj))
+
+                wait_for_dc_app_pods_to_reach_running_state(dc_pod_objs)
+
+                for pod_obj in dc_pod_objs:
+                    pod_obj.run_io(
+                        storage_type='fs', size='3G', runtime='60',
+                        fio_filename=f'{pod_obj.name}_io'
+                    )
+
+        # Add capacity
+        osd_size = storage_cluster.get_osd_size()
+        count = storage_cluster.add_capacity(osd_size)
+        pod = OCP(
+            kind=constants.POD, namespace=config.ENV_DATA['cluster_namespace']
+        )
+        pod.wait_for_resource(
+            timeout=300,
+            condition=constants.STATUS_RUNNING,
+            selector='app=rook-ceph-osd',
+            resource_count=count * 3
+        )
+        assert ceph_health_check(), "New OSDs failed to reach running state"
+
+        cluster = CephCluster()
+
+        # Get rebalance status
+        rebalance_status = cluster.get_rebalance_status()
+        if rebalance_status:
+            time_taken = cluster.time_taken_to_complete_rebalance()
+            logger.info(f"The time taken to complete rebalance {time_taken}")
+
+        worker_nodes = get_typed_nodes(node_type='worker')
+
+        node_obj = nodes()
```

Comment:
nodes() is a fixture. The correct usage of it should be with injecting it into the test function. See reference in https://github.com/red-hat-storage/ocs-ci/blob/8dd733926df64676c136be3c76e62239443ac733/tests/manage/z_cluster/nodes/test_disk_failures.py#L143

Reply:
Done